### ISSUE TITLE
Beatmap difficulties are showing out of order

### ISSUE THREAD

**Issue by Jaren8r** [ppy/osu#20245]
### Type

Cosmetic

### Bug description

I was looking through my beatmaps, and I noticed that the difficulties of my maps are sorted randomly, even with sort by at the top being set to difficulty (which I don't think influences it, but I tried anyways). I also noticed that the difficulty color circles that show on the map are all grey, and show `-` as the star rating.

### Screenshots or videos

![osu_2022-09-11_11-46-37](https://user-images.githubusercontent.com/8620569/189536765-b7f8dcb0-4e1e-4829-8c17-fa1247ff69eb.png)

### Version

2022.911.0-lazer

### Logs

[performance.log](https://github.com/ppy/osu/files/9542950/performance.log)
[performance-audio.log](https://github.com/ppy/osu/files/9542951/performance-audio.log)
[performance-draw.log](https://github.com/ppy/osu/files/9542952/performance-draw.log)
[performance-input.log](https://github.com/ppy/osu/files/9542953/performance-input.log)
[performance-update.log](https://github.com/ppy/osu/files/9542954/performance-update.log)
[runtime.log](https://github.com/ppy/osu/files/9542955/runtime.log)
[updater.log](https://github.com/ppy/osu/files/9542956/updater.log)
[database.log](https://github.com/ppy/osu/files/9542957/database.log)
[network.log](https://github.com/ppy/osu/files/9542958/network.log)

**Comment by LeNitrous** [comment]
Isn't this because of a recent difficulty calc bump? It should be fine after recalculating as it happened to me earlier as well.

**Comment by Jaren8r** [comment]
Ah whoops, It's likely that. I scrolled all the way down and I'm seeing beatmaps regaining color.

**Comment by Jaren8r** [comment]
I still think this should be addressed, as recalculating can take quite a bit of time if you have a lot of maps (still running for me...). It makes it impossible to search for maps by difficulty unless I wait for a long time for it to process on the song select screen. Maybe have it use the pre-recalc difficulty until it's calculated..?

**Comment by smoogipoo** [comment]
Yes it is indeed due to the diffcalc bump. osu!stable uses negative values specifically for this reason whereas lazer only sets -1 and calls it a day.

**Comment by peppy** [comment]
I think we discussed in the original implementation, but in case anyone jumps in to address this, we don't want to be using negative values this time around. A separate property tracking the updated state should work well.
